### PR TITLE
[Model Update]: JustInSequencePart 3.0.0 

### LINKED ISSUE #560: [Model Update]: JustInSequencePart 3.0.0
## Update Reason
Based on the continous development of the Industry Core, there are a new data, that needs to be integrated in the **JustInSequencePart** aspect model:

- Integration of the new shared Classifcation model (#520)

Additionally there are a few adaptation to be done in general:
- Permit to add date information excluding time 
``` 
^\d{4}-\d{2}-\d{2}(T\d{2}:\d{2}:\d{2}(\.\d{1,6})?(Z|[+-]\d{2}:\d{2}))?$
``` 
- Change (shared) partSiteInformation to be a child-property of the manufacturerInformation
- Include RegEx for the local identifier key 
``` 
^(manufacturerId|jisNumber|jisCallDate|parentOrderNumber|customKey:\w+)$
```

## Relevant Standards
- CX-0127 Industry Core: Part Instance 2.0.0 (if merged before 29.02.24)

## Example Data
```json
{
  "catenaXId": "urn:uuid:580d3adf-1981-44a0-a214-13d6ceed9379",
  "localIdentifiers": [
    {
      "value": "12345678ABC",
      "key": "jisNumber"
    }
  ],
  "manufacturingInformation": {
    "date": "2022-02-04T14:48:54",
    "country": "HUR",
    "sites": [
      {
        "catenaXsiteId": "BPNS1234567890ZZ",
        "function": "production"
      }
    ]
  },
  "partTypeInformation": {
    "manufacturerPartId": "123-0.740-3434-A",
    "customerPartId": "PRT-12345",
    "nameAtManufacturer": "Mirror left",
    "nameAtCustomer": "side element A",
    "partClassification": [
      {
        "classificationStandard": "IEC",
        "classificationID": "61360- 2:2012 ",
        "classificationDescription": "Standard data element types with associated classification scheme for electric components."
      }
    ]
  }
}
```

## MS1 Criteria

- [x] The model that should be updated exists
- [x] The referenced use case exists
- [x] The potential updates are discussed with all stakeholders
- [x] A decision on versioning was made in consensus with the affected parties [according to the guidelines](https://github.com/eclipse-tractusx/sldt-semantic-models/blob/main/documentation/GOVERNANCE.md#versioning).
- [x] The potential update will be either 
  - [ ] backward compatible or 
  - [x] a (chain of) migration strategy(-ies) from all non-deprecated previous versions to the new model will be developed
- [x] A modeling steward has been assigned

### PR DESCRIPTION
**Pull request by johannsvarela**
## Description
Based on the continous development of the Industry Core, there are a new data, that needs to be integrated in the **JustInSequencePart** aspect model. Therefore these extensions / adaptations were made:

- Integration of the new shared Classifcation model
- Permit to add date information excluding time
- Change (shared) partSiteInformation to be a child-porperty of the manufacturerInformation
- Include RegEx for the local identifier key

Closes #560 --> Deprecation of the old version (#614) should be done after this PR was merged!

## MS2 Criteria
(to be filled out by PR reviewer)
- [x] the model **validates** with the SAMM SDS SDK in the version specified in the Readme.md of this repository by the time of the MS2 check  (e.g., 'java -jar samm-cli.jar aspect \<path-to-aspect-model\> validate ). The  SAMM CLI is available [here](https://eclipse-esmf.github.io/esmf-developer-guide/tooling-guide/samm-cli.html) and in [GitHub](https://github.com/eclipse-esmf/esmf-sdk/releases/tag/v2.3.2)
- [x] use **Camel-Case** (e.g., "MyModelElement" or "TimeDifferenceGmtId", when in doubt follow https://google.github.io/styleguide/javaguide.html#s5.3-camel-case)
- [x] the identifiers for all model elements **start with a capital letter** except for properties
- [x] the identifier for **properties starts with a small letter**
- [x] all model elements **at least contain the fields "preferred name" and "description"** in English language. The description must be comprehensible. It is not required to write full sentences but style should be consistent over the whole model
- [x] Property and the referenced Characteristic should not have the same name
- [x] the versioning in the URN **follows semantic versioning**, where minor version bumps are backwards compatible and major version bumps are not backwards compatible. 
- [x] use **abbreviations only when necessary** and if these are sufficiently common
- [x] **avoid redundant prefixes in property names** (consider adding properties to an enclosing Entity or even adapt the namespace of the model elements, e.g., instead of having two properties `DismantlerId` and `DismantlerName` use an Entity `Dismantler` with the properties `name` and `id` or use a URN like `io.catenax.dismantler:0.0.1`)
- [x] fields `preferredName` and `description` are not the same
- [x] **`preferredName` should be human readable** and follow normal orthography (e.g., no camel case but normal word separation)
- [x] name of aspect is singular except if it only has one property which is a Collection, List or Set. In theses cases, the aspect name is plural.
- [x] units are referenced from the SAMM unit catalog whenever possible
- [x] **use constraints** to make known constraints from the use case explicit in the aspect model 
- [x] when relying on **external standards**, they are referenced through a **"see"** element
- [x] all properties with an [simple type](https://eclipse-esmf.github.io/samm-specification/2.0.0/datatypes.html) have an example value
- [x] metadata.json exists with status "release"
- [x] generated json schema validates against example json payload
- [x] file RELEASE_NOTES.md exists and contains entries for proposed model changes 
- [x] all contributors to this model are mentioned in copyright header of model file

## MS3 Criteria
(to be filled out by semantic modeling team before merge to main-branch)
- [x] All required reviewers have approved this PR (see reviewers section)
- [x] The new aspect (version) will be implemented by at least one data provider
- [x] The new aspect (version) will be consumed by at least one data consumer
- [x] There exists valid test data
- [x] In case of a new (incompatible) major version to an existing version, a migration strategy has been developed
- [x] The model has at least version '1.0.0'
- [x] If a previous model exists, model deprecation has been checked for previous model
- [x] The release date in the Release Note is set to the date of the MS3 approval

### CODE REVIEW

**Comment by johannsvarela** [comment]
FYI: @jacewski-bosch @BenediktMuellerIML @jonbckr

**Comment by johannsvarela** [comment]
modeling_team

File: io.catenax.just_in_sequence_part/3.0.0/JustInSequencePart.ttl [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+@prefix samm: <urn:samm:org.eclipse.esmf.samm:meta-model:2.1.0#>.
+@prefix samm-c: <urn:samm:org.eclipse.esmf.samm:characteristic:2.1.0#>.
+@prefix samm-e: <urn:samm:org.eclipse.esmf.samm:entity:2.1.0#>.
+@prefix unit: <urn:samm:org.eclipse.esmf.samm:unit:2.1.0#>.
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>.
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#>.
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#>.
+@prefix : <urn:samm:io.catenax.just_in_sequence_part:3.0.0#>.
+@prefix ext-built: <urn:samm:io.catenax.shared.part_site_information_as_built:1.0.0#>.
```
Comment (agg3fe) [inline io.catenax.just_in_sequence_part/3.0.0/JustInSequencePart.ttl:34]: part_site_information_as_built 2.0.0 required

File: io.catenax.just_in_sequence_part/3.0.0/JustInSequencePart.ttl [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+@prefix ext-classification: <urn:samm:io.catenax.shared.part_classification:1.0.0#>.
+@prefix ext-uuid: <urn:samm:io.catenax.shared.uuid:1.0.0#>.
```
Comment (agg3fe) [inline io.catenax.just_in_sequence_part/3.0.0/JustInSequencePart.ttl:36]: uuid 2.0.0

File: io.catenax.just_in_sequence_part/3.0.0/JustInSequencePart.ttl [hunk 3]
```diff
@@ -0,0 +1,207 @@
+#######################################################################
+# Copyright(c) 2024 BASF SE
```
Comment (agg3fe) [inline io.catenax.just_in_sequence_part/3.0.0/JustInSequencePart.ttl:2]: 2023,2024

File: io.catenax.just_in_sequence_part/RELEASE_NOTES.md [hunk 4]
```diff
@@ -1,6 +1,15 @@
 # Changelog
 All notable changes to this model will be documented in this file.
 
+## [3.0.0] 2024-02-05
+### Added
+- Integration of the new shared PartClassifcation 1.0.0 aspect model as child-property of the PartTypeInformation
```
Comment (agg3fe) [inline io.catenax.just_in_sequence_part/RELEASE_NOTES.md:6]: typo `PartClassifcation`

File: io.catenax.just_in_sequence_part/RELEASE_NOTES.md [hunk 5]
```diff
@@ -1,6 +1,15 @@
...
+- Added possibility to add date information excluding time
+- Include additional RegEx values for the local identifier key
+
+### Changed
+- Change (shared) partSiteInformation to be a child-porperty of the manufacturerInformation
```
Comment (agg3fe) [inline io.catenax.just_in_sequence_part/RELEASE_NOTES.md:11]: typo `porperty`

**Review by agg3fe — CHANGES_REQUESTED** [review]
please check my comments

**Comment by johannsvarela** [comment]
> please check my comments

@agg3fe Changes were made like suggested. Please review this PR again! :)

**Review by agg3fe — APPROVED** [review]
LGTM

**Pull request merged**